PWGCF/TwoParticleCorrelations/Tasks/r2p2-4-id.cxx

```diff
@@ -0,0 +1,437 @@
+    flag = pidarray<decltype(track1)>[p2](track1);
+    if (flag == true) {
+      histos.fill(HIST("h2d_n1_tof2"), track1.pt(), track1.beta());
+      histos.fill(HIST("h2d_n1_tpc2"), track1.pt(), track1.tpcSignal());
+      histos.fill(HIST("h1d_n1_phi2"), track1.phi());
+      histos.fill(HIST("h1d_n1_eta2"), track1.eta());
+      sign1 = (track1.sign() + 1) / 2;
+      h1d_1p[1][sign1]->Fill(track1.pt(), 1.0 / (2.0 * constants::math::PI * track1.pt()));
+      h2d_1p[1][sign1][0]->Fill(track1.eta(), track1.phi());
+      h2d_1p[1][sign1][1]->Fill(track1.eta(), track1.phi(), track1.pt());
+    }
+    //-------------------------------------------------------------------------
+    //-----PROTON PID Particle1------------------------------------------------
+    if (!pidarray<decltype(track1)>[p1](track1))
+      continue;
+    histos.fill(HIST("h2d_n1_tof1"), track1.pt(), track1.beta());
+    histos.fill(HIST("h2d_n1_tpc1"), track1.pt(), track1.tpcSignal());
+    histos.fill(HIST("h1d_n1_phi1"), track1.phi());
+    histos.fill(HIST("h1d_n1_eta1"), track1.eta());
+    sign1 = (track1.sign() + 1) / 2;
+    h1d_1p[0][sign1]->Fill(track1.pt(), 1.0 / (2.0 * constants::math::PI * track1.pt()));
+    h2d_1p[0][sign1][0]->Fill(track1.eta(), track1.phi());
+    h2d_1p[0][sign1][1]->Fill(track1.eta(), track1.phi(), track1.pt());
+    //-------------------------------------------------------------------------
+    etabin1 = (track1.eta() + 0.8) * 15;
+    phibin1 = 36 * track1.phi() / (2 * constants::math::PI);
+    if ((etabin1 < 0) || (etabin1 >= 24) || (phibin1 < 0) || (phibin1 >= 36))
+      continue;
+    mult++;
+    //-------------------------------------------------------------------------
+    for (auto track2 : tracks) {
+      //-----Two Particle Distribution-----------------------------------------
+      if (track1.index() == track2.index())
+        continue; // No auto-correlations
+      //-----KAON PID Particle 2-----------------------------------------------
+      if (!pidarray<decltype(track2)>[p2](track2))
+        continue;
+      //-----------------------------------------------------------------------
+      etabin2 = (track2.eta() + 0.8) * 15;
+      phibin2 = 36 * track2.phi() / (2 * constants::math::PI);
+      if ((etabin2 < 0) || (etabin2 >= 24) || (phibin2 < 0) || (phibin2 >= 36))
+        continue;
+      sign2 = (track2.sign() + 1) / 2;
+      h2d_2p[sign1][sign2][0]->Fill(36 * etabin1 + phibin1 + 0.5, 36 * etabin2 + phibin2 + 0.5);
+      h2d_2p[sign1][sign2][1]->Fill(36 * etabin1 + phibin1 + 0.5, 36 * etabin2 + phibin2 + 0.5, track2.pt());
+      h2d_2p[sign1][sign2][2]->Fill(36 * etabin1 + phibin1 + 0.5, 36 * etabin2 + phibin2 + 0.5, track1.pt());
+      h2d_2p[sign1][sign2][3]->Fill(36 * etabin1 + phibin1 + 0.5, 36 * etabin2 + phibin2 + 0.5, track1.pt() * track2.pt());
+      //-----------------------------------------------------------------------
+    }
+  }
+  histos.fill(HIST("h1i_n1_multPM"), mult);
+}
```

Comment:
Please, consider that if the processing of the singles is considered in an independent function, most probably the processing of the pairs can be done without the need of replicating the like and unlike codes